openverse_catalog/dags/common/requester.py

```diff
@@ -5,6 +5,16 @@
 import requests
 
 
+# pytest_socket will not be available in production, so we must create a shim for
+# that specific exception so this code does not error out.
+try:
+    from pytest_socket import SocketConnectBlockedError
```

Comment:
So, it's saying "if we already imported `pytest_socket` somewhere else, great, use that, otherwise use this nothing thing", right? ("shim" is new to me, but it means the nothing thing and/or the wrapper around it?) That's cool!

Reply:
Great question! Usually with python imports, it's necessary to import what you need in whatever file it's required in. Python then manages which modules have already been imported by using a [module cache](https://docs.python.org/3/reference/import.html#the-module-cache). So if we `import foo` in file `a.py` which gets loaded first, and we import it in `b.py` which gets loaded second, the second `import foo` will just reference the module that was imported as part of `a.py`.

This is a bit of a different case. The Docker images that we build for local development and for production are different. We don't need the development dependencies (like pytest, code linters, etc) when we're running Airflow in production; they're only useful when we're working on the project on our machines. As such, the images use different requirements files when being built (prod [arg](https://github.com/WordPress/openverse-catalog/blob/8754cb63c075e2045b91af7bee278566d930286b/docker/airflow/Dockerfile#L4), [file](https://github.com/WordPress/openverse-catalog/blob/8754cb63c075e2045b91af7bee278566d930286b/requirements_prod.txt)) (dev [arg](https://github.com/WordPress/openverse-catalog/blob/8754cb63c075e2045b91af7bee278566d930286b/docker-compose.override.yml#L57), [file](https://github.com/WordPress/openverse-catalog/blob/8754cb63c075e2045b91af7bee278566d930286b/requirements_dev.txt)).

All that to say: when we're running in production, the python interpreter there _won't have `pytest_socket` available at all_. While we want that exact exception for local testing, we can just create a [shim](https://en.wikipedia.org/wiki/Shim_(computing)) (or sham 😜) exception which is used in its place in production. In other words, "if we don't have `pytest_socket` installed then just reference this empty exception class instead". All this is contingent on the fact that this exception **will not and cannot** be raised in production (since we don't have the library anyway!).

Reply:
Awesome! Thank you so much @AetherUnbound !